File: client/android/BigFoxAndroid/app/src/main/java/vn/com/onesoft/bigfox/io/core/BFLogger.java
package vn.com.onesoft.bigfox.io.core;

import android.util.Log;

import vn.com.onesoft.bigfox.io.message.BaseMessage;

/**
 * Created by phamquan on 8/16/15.
 */
public class BFLogger {private static BFLogger _instance = null;

    public static BFLogger getInstance() {
        if (_instance == null) {
            _instance = new BFLogger();
        }
        return _instance;
    }

    public void debug(Object obj){
        if (obj instanceof BaseMessage &&  ((BaseMessage) obj).isCore())
            return;
        Log.d("BigFox", obj.toString());
    }

    public void info(Object obj) {
        if (obj instanceof BaseMessage &&  ((BaseMessage) obj).isCore())
            return;
        Log.i("BigFox", obj.toString());
    }

    public void error(Object obj, Throwable t) {
        Log.e("BigFox", obj.toString(), t);
    }

    public void error(Object obj) {
        Log.e("BigFox", obj.toString());
    }
}

File: client/ios/BigFoxClient/BigFoxClient/message/BFLogger.m
//
//  BFLogger.m
//  BigFoxClient
//
//  Created by QuyenNX on 9/4/15.
//  Copyright (c) 2015 QuyenNX. All rights reserved.
//

#import "BFLogger.h"
#import "BigFoxUtils.h"

@implementation BFLogger
static BFLogger* _instance;
+ (BFLogger*) getInstance {
    if (_instance == nil) {
        _instance = [[BFLogger alloc] init];
    }
    return _instance;
}

- (void) log : (NSString*) str {
    NSLog(str);
}
- (void) info:(NSObject *)obj {
    NSLog([NSString stringWithFormat:@"%@: %@",NSStringFromClass([obj class]), [BigFoxUtils toString:obj]]);
}

- (void) error:(NSObject *)obj {
     NSLog([NSString stringWithFormat:@"error at object: %@",[BigFoxUtils toString:obj]]);
}

-(void) error:(NSObject *)obj :(NSException *)ex {
    NSLog([NSString stringWithFormat:@"%@ %@", [BigFoxUtils toString:obj], [ex description]]);
}

-(void) logChat:(char *)data : (int) length{
    NSString* result = @"";
    for ( int i = 0; i < length; i++) {
        result = [result stringByAppendingString:[NSString stringWithFormat:@"%d",data[i]]];

    }
            NSLog(result);
}
@end
